velox/functions/prestosql/types/tests/BingTileTypeTest.cpp

```diff
@@ -40,4 +40,36 @@ TEST_F(BingTileTypeTest, serde) {
   testTypeSerde(BINGTILE());
 }
 
+TEST_F(BingTileTypeTest, packing) {
+  uint64_t tile_x123_y111_z7 = bingTileCoordsToInt(123, 111, 7);
+  ASSERT_TRUE(isBingTileIntValid(tile_x123_y111_z7));
+  ASSERT_EQ(bingTileVersion(tile_x123_y111_z7), 0);
+  ASSERT_EQ(bingTileZoom(tile_x123_y111_z7), 7);
+  ASSERT_EQ(bingTileX(tile_x123_y111_z7), 123);
+  ASSERT_EQ(bingTileY(tile_x123_y111_z7), 111);
+
+  uint64_t tile_z0 = bingTileCoordsToInt(0, 0, 0);
+  ASSERT_TRUE(isBingTileIntValid(tile_z0));
+  ASSERT_EQ(bingTileVersion(tile_z0), 0);
+  ASSERT_EQ(bingTileZoom(tile_z0), 0);
+  ASSERT_EQ(bingTileX(tile_z0), 0);
+  ASSERT_EQ(bingTileY(tile_z0), 0);
+
+  // Error cases
+  uint64_t tile_v1 = (uint64_t)1 << (63 - 5);
+  ASSERT_FALSE(isBingTileIntValid(tile_v1));
+  ASSERT_EQ(bingTileInvalidReason(tile_v1), "Version 1 not supported");
+
+  uint64_t tile_z24 = bingTileCoordsToInt(0, 0, 24);
+  ASSERT_FALSE(isBingTileIntValid(tile_z24));
+  ASSERT_EQ(
+      bingTileInvalidReason(tile_z24), "Zoom 24 is greater than max zoom 23");
+
+  uint64_t tile_x0_y1_z0 = bingTileCoordsToInt(0, 1, 0);
+  ASSERT_FALSE(isBingTileIntValid(tile_x0_y1_z0));
+  ASSERT_EQ(
+      bingTileInvalidReason(tile_x0_y1_z0),
```

Comment:
do we care about the zero bits (5-6 and 36-40 bit positions)? If not, can we add a test case to show that its valid for completeness

Reply:
They are unused in V1 (which likely is the only version): nothing depends on them. Would you want a test showing they are zero?

Reply:
got it, in that case we can skip adding a test.